src/beliefpropagation/beliefpropagation_schedule.jl

```diff
@@ -0,0 +1,40 @@
+default_edge_sequence_alg() = "forest_cover"
+
+@traitfn default_bp_niters(g::NamedGraph::(!IsDirected)) = is_tree(g) ? 1 : nothing
+function default_bp_niters(g::AbstractGraph)
+  return default_bp_niters(undirected_graph(underlying_graph(g)))
+end
+
+function edge_sequence(g::NamedGraph; alg=default_edge_sequence_alg())
+  return edge_sequence(alg, g)
+end
+
+function edge_sequence(g::AbstractGraph; alg=default_edge_sequence_alg())
+  return edge_sequence(Algorithm(alg), undirected_graph(underlying_graph(g)))
+end
+
+function edge_sequence(alg::Algorithm, g::AbstractGraph; kwargs...)
+  return edge_sequence(alg, g, kwargs...)
+end
+
+@traitfn function edge_sequence(
+  ::Algorithm"forest_cover",
+  g::NamedGraph::(!IsDirected);
```

Comment:
```suggestion
  g::::(!IsDirected);
```

Reply:
Currently `NamedGraphs.forest_cover` (a function called by this functions) acts only on an `AbstractNamedGraph` ?

Reply:
So this would raise an error if you passed it an undirected `datagraph` or something